internal/quaiapi/api.go

```diff
@@ -926,13 +926,13 @@ type RPCTransaction struct {
 func newRPCTransaction(tx *types.Transaction, blockHash common.Hash, blockNumber uint64, index uint64, baseFee *big.Int) *RPCTransaction {
 	// Determine the signer. For replay-protected transactions, use the most permissive
 	// signer, because we assume that signers are backwards-compatible with old
-	// transactions. For non-protected transactions, the homestead signer signer is used
+	// transactions. For non-protected transactions, the signer is used
 	// because the return value of ChainId is zero for those transactions.
 	var signer types.Signer
 	if tx.Protected() {
 		signer = types.LatestSignerForChainID(tx.ChainId())
 	} else {
-		signer = types.HomesteadSigner{}
+		types.LatestSigner(params.RopstenChainConfig)
```

Comment:
why is this RopstenChainConfig.

Reply:
Good catch. However if the tx is *not* replay preotected it should probably be rejected, right?

Reply:
yeah, we should probably use MainnetChainConfig as default. Ideally would be nice to switch the signer based on the type of chain you are running.

Reply:
No, I mean in the else the tx should be rejected completely

Reply:
Update: Created empty tx in the case of unprotected tx

Reply:
like it. thanks.